docs/adrs/adr-001-globalfee.md

```diff
@@ -0,0 +1,147 @@
+
+To address this, the `CombinedFeeRequirement` is split into zero and non-zero coins, forming `nonZeroCoinFeesReq` and `zeroCoinFeesDenomReq`. Similarly, the paid fees (feeCoins) are split into `feeCoinsNonZeroDenom` and `feeCoinsZeroDenom`, based on the denominations of `nonZeroCoinFeesReq` and `zeroCoinFeesDenomReq`.
+
+The split enable checking `feeCoinsNonZeroDenom` against `nonZeroCoinFeesReq`, and `feeCoinsZeroDenom` against
+`zeroCoinFeesDenomReq`. In the check of `feeCoinsNonZeroDenom` against `nonZeroCoinFeesReq`, the sdk coins' methods can be used since zero coins are removed from the `nonZeroCoinFeesReq`, while in the check `feeCoinsZeroDenom` against `zeroCoinFeesDenomReq`, only denoms need to be checked.
+
+```go
+	nonZeroCoinFeesReq, zeroCoinFeesDenomReq := getNonZeroFees(feeRequired)
+
+	// feeCoinsNonZeroDenom contains non-zero denominations from the feeRequired
+	// feeCoinsNonZeroDenom is used to check if the fees meets the requirement imposed by nonZeroCoinFeesReq
+	// when feeCoins does not contain zero coins' denoms in feeRequired
+	feeCoinsNonZeroDenom, feeCoinsZeroDenom := splitCoinsByDenoms(feeCoins, zeroCoinFeesDenomReq)
+
+```
+
+```go 
+	if !feeCoinsNonZeroDenom.IsAnyGTE(nonZeroCoinFeesReq) {
+		return ctx, sdkerrors.Wrapf(sdkerrors.ErrInsufficientFee, "insufficient fees; got: %s required: %s", feeCoins.String(), feeRequired.String())
+	}
+```
+An example of coins split in fee antehandler:
+`globalfee=[1photon, 0uatom, 1stake]`\
+`local min-gas-prices=[0.5stake]`
+get `combinedFeeRequirement=[1photon, 0uatom, 1stake]`,
+split the `combinedFeeRequirement` into `nonZeroCoinFeesReq=[0uatom]`, and `nonZeroCoinFeesReq=[1photon, 1stake]`
+if `paidFee=[1uatom, 0.5photon]`,
+split the paidFee into `feeCoinsZeroDenom=[1uatom]` (the same denom as zero coins in `combinedFeeRequirement`), and `feeCoinsNonZeroDenom=[0.5stake]`
+then `feeCoinsZeroDenom=[1uatom]` is checked by `nonZeroCoinFeesReq=[1photon, 1stake]`.
+
+```mermaid
+---
+title: Fee Requirements and Fee Splits
+---
+flowchart TD
+	subgraph feeReq
+    CombinedFeeRequirement-->zeroCoinFeesDenomReq;
+	CombinedFeeRequirement-->nonzeroCoinFeesDenomReq;
+	end
+	subgraph feeCoin
+	feeCoins-->feeCoinsZeroDenom;
+	feeCoins-->feeCoinsNonZeroDenom;
+	end
+
+	feeCoinsZeroDenom-.feeCheck-Zero.->zeroCoinFeesDenomReq;
+	feeCoinsNonZeroDenom-.feeCheck-NonZero.->nonzeroCoinFeesDenomReq;
+```
+plese note, `feeCoins` does not contain zero coins, the feeCoins are split to according to the denoms in `zeroCoinFeesDenomReq` or in `nonZeroCoinFeesDenomReq`, if feeCoins contains coins not in both  `zeroCoinFeesDenomReq` and `nonZeroCoinFeesDenomReq`, the transaction should be rejected.
```

Comment:
```suggestion
Please note that `feeCoins` does not contain zero coins. The fee coins are split according to the denoms in `zeroCoinFeesDenomReq` or `nonZeroCoinFeesDenomReq`. If feeCoins contains coins not in both `zeroCoinFeesDenomReq` and `nonZeroCoinFeesDenomReq`, the transaction should be rejected.
```
This sentence should be exentended to clairify that if `len(feeCoins)==0` and  `len(zeroCoinFeesDenomReq)` the transaction pass.